src/utils/rewards.js

```diff
@@ -75,19 +77,174 @@ export const getPassiveUserAPY = async (userVeOcean, lockedOcean) => {
   const veOceanSupply = await getTotalVeSupply();
   let curEpoch = getEpoch();
   let passiveRewards =
-    import.meta.env.VITE_VE_SUPPORTED_CHAINID != "1" ? 20 : curEpoch?.streams[0]?.substreams[0]?.rewards;
+    import.meta.env.VITE_VE_SUPPORTED_CHAINID != "1"
+      ? 20
+      : curEpoch?.streams[0]?.substreams[0]?.rewards;
   const rewards = (passiveRewards / veOceanSupply) * userVeOcean;
   const wpr_passive = rewards / lockedOcean;
   return convertWPRtoAPY(wpr_passive);
+};
+
+export const getMsDelta = (unlockDate) => {
+  var today = moment.utc();
+  var unlockDate = moment.utc(unlockDate);
+  return unlockDate.diff(today);
 }
 
+export const getVotingPower = (msDelta, getMaxDate, totalAmount) => {
+  return ((msDelta / getMaxDate().diff(moment.utc())) * totalAmount).toFixed(3);
+};
+
+const calculateFeeForPeriod = (fees, unlockDate) => {
+  // Assuming 'fees' contains fee values for different actions (lock, claim, etc.)
+  // and 'unlockDate' is the moment object of the unlock date
+
+  const numberOfClaims = calculateNumberOFClaims(unlockDate); // Function from your 'rewards.js'
+  const feePerClaim = fees.claim * numberOfClaims;
+  const feeForLock = fees.lock;
+  const totalFeeForPeriod = feePerClaim + feeForLock; // Summing up fees for lock and claims
+
+  return totalFeeForPeriod;
+};
+
+const calculateAPYBasedOnTotalRewards = (principal, totalRewards) => {
+  if (principal === 0) return 0; // Avoid division by zero
+
+  // Calculate the APY based on the total rewards and the principal
+  const apy = (totalRewards / principal) * 100; // Convert the ratio to a percentage
+
+  return apy;
+};
+
+export const calculateCompoundInterestWithFees = async ({
+  msDelta,
+  getMaxDate,
+  lockedOceanAmount,
+  formAmount,
+  formUnlockDate,
+  fees,
+  displayedAPY,
+  totalVeOceanSupply,
+  compounds,
+}) => {
+  const formatApyForDisplay = (apy, rewards) => {
+    return { apy: apy, profit: rewards };
+  };
+
+  const startPrincipal =
+    formAmount > 0
+      ? formAmount + parseFloat(lockedOceanAmount)
+      : parseFloat(lockedOceanAmount);
+
+  let calculatedVotingPower = getVotingPower(msDelta, getMaxDate, startPrincipal);
+
+  if (
+    calculatedVotingPower <= 0 ||
+    (!lockedOceanAmount && formAmount <= 0) ||
+    !fees
+  ) {
+    displayedAPY = formatApyForDisplay(0, 0);
+    return;
+  }
+
+  let principal = startPrincipal;
+
+  let totalRewards = 0;
+
+  let totalVotingPower = 0;
+  for (let i = 0; i < compounds; i++) {
+    let tempVotingPower = getVotingPower(
+      msDelta / compounds,
```

Comment:
we shouldn't divide the lock time by compounds because it's going to result wrong veOCEAN amount

Reply:
Firstly, It is moved to this part:
https://github.com/oceanprotocol/df-web/pull/686/files#diff-b357279e0988e4f14f3ba52534dbc149574f1aecaab34c70dcbe95276e863124R30

However, this modification doesn't negatively impact the calculation. To imagine, let's consider a scenario where maxDate equals 'X'.

When we divide 'X' by 3, for instance, we essentially have three separate segments, each representing `(1/X) * totalAmount`. this is expressed as 3 * (1/X) * r. On the other hand, if we didn't divide and directly used `(3/X) * r`, the outcome remains the same.

The division of the lock time by the number of compounds is crucial because the `totalAmount` changes with each compounding event. Therefore, to accurately account for these changes over time, the lock time must be divided accordingly